shopinvader_invoice/tests/test_invoice_service.py

```diff
@@ -0,0 +1,185 @@
+# -*- coding: utf-8 -*-
+# Copyright 2019 ACSONE SA/NV (<http://acsone.eu>)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from odoo import fields
+from odoo.addons.shopinvader.tests.common import CommonCase
+
+
+class TestInvoiceService(CommonCase):
+    """
+    Tests for
+    """
+
+    def setUp(self, *args, **kwargs):
+        super(TestInvoiceService, self).setUp(*args, **kwargs)
+        self.invoice_obj = self.env["account.invoice"]
+        self.partner = self.env.ref("base.res_partner_2").copy()
+        self.product = self.env.ref("product.product_product_4")
+        self.precision = 2
+        with self.work_on_services(partner=self.partner) as work:
+            self.service = work.component(usage="invoice")
+        with self.work_on_services(
+            partner=self.backend.anonymous_partner_id
+        ) as work:
+            self.service_guest = work.component(usage="invoice")
+
+    def _get_selection_label(self, invoice, field):
+        """
+        Get the translated label of the invoice selection field
+        :param invoice: account.invoice recordset
+        :param field: str
+        :return: str
+        """
+        technical_type = invoice[field]
+        type_dict = dict(
+            invoice._fields.get(field)._description_selection(invoice.env)
+        )
+        return type_dict.get(technical_type, technical_type)
+
+    def _check_data_content(self, data, invoices):
+        """
+        Check data based on given invoices
+        :param data: list
+        :param invoices: account.invoice recordset
+        :return: bool
+        """
+        # To have them into correct order
+        invoices = invoices.search(
+            [
+                ("id", "in", invoices.ids),
+                # Invoice must be paid to be in data
+                ("state", "in", ["open", "paid"]),
+            ]
+        )
+        self.assertEquals(len(data), len(invoices))
+        for current_data, invoice in zip(data, invoices):
+            state_label = self._get_selection_label(invoice, "state")
```

Comment:
you should test w/ hardcoded value for the labels w/out duplicating that method.

Reply:
I can't "hardcode" a value because this `_check_data_content` function compare given `data` with given `invoices`.